src/codeu/chat/client/commandline/Chat.java

```diff
@@ -46,11 +46,19 @@ public Chat(Context context) {
   // the system wants to exit, the function will return false.
   //
   public boolean handleCommand(String line) {
-
+/*
     final Scanner tokens = new Scanner(line.trim());
-
     final String command = tokens.hasNext() ? tokens.next() : "";
-
+*/
+  final List<String> args = new ArrayList<>();
+  final Tokenizer tokenizer = new Tokenizer(line);
+  for(String token = tokenizer.next(); token != null; token = tokenizer.next())
```

Comment:
You also need to wrap this in a try-catch block or throw an exception, because `tokenizer.next()` can throw an IOException.